test/hotspot/gtest/runtime/test_os_linux.cpp

```diff
@@ -455,7 +455,11 @@ TEST_VM(os_linux, decoder_get_source_info_valid_truncated) {
   ASSERT_TRUE(Decoder::get_source_info(valid_function_pointer, buf, 7, &line));
   ASSERT_TRUE(buf[7 - 1] == '\0');
   ASSERT_TRUE(buf[7] == 'X');
-  ASSERT_TRUE(strcmp(buf, "jni.cp") == 0);
+  // Clang emits a relative file path for debug builds while in product builds it only emits the filename itself.
+  // GCC only emits the filename.
+  // This gives us either "jni.cp" or "src/ho". In the latter case, we strip the path prefix to get to the actual
+  // filename which, however, is useless in this case - we get "ho".
+  ASSERT_TRUE(strcmp(buf, "jni.cp") == 0 || strcmp(buf, "ho") == 0);
```

Comment:
I did not try it out, but since we already stripped the prefix before printing the file name, why can we get "ho" here?

Reply:
I updated the comment to make it more clear. Due to the small buffer, we only read "src/ho" instead of "src/hotspot/share...". And since we are also calling `strip_path_prefix`, we strip "src/" and we get "ho" as filename which is not actually a filename.